no_waste/recipes/views.py

```diff
@@ -1,112 +1,122 @@
-    def __init__(self, title, picture, ingredients,
-        missing_ingredients_count, carbs, proteins, fats, calories, sourceurl):
-
-        self.title_en = title
-        self.title_pt = ""
-        self.picture = picture
-        self.ingredients_en = ingredients
-        self.ingredients_pt = []
-        self.missing_ingredients_count = missing_ingredients_count
-        self.carbs = carbs
-        self.proteins = proteins
-        self.fats = fats
-        self.calories = calories
-        self.sourceurl = sourceurl
-
-    def translator(self, text):
-        """This method translates text in English to Portuguese"""
+    def get_translation(self, text):
+        """This method translates English to Portuguese"""
         translator = Translator()
-        translation = translator.translate(text=text, src="en", dest="pt")
+        translation = translator.translate(text=text, src="en", dest="pt").text
+
         return translation
-    
-    def get_translation(self):
-        """Translates title and ingredients to Portuguese and assigns it to corresponding attribute"""
-        self.title_pt = self.translator(self.title_en).text
-        self.ingredients_pt = [self.translator(ingredient).text for ingredient in self.ingredients_en]
-
-
-
-
-def search_recipes_view(request):
-    """This function renders a list with the recipes resulting from the user query"""
-
-    #Gets query data
-    query_dict = request.GET
-    ingredients_to_include = query_dict["ingredients_to_include"]
-    ingredients_to_exclude = query_dict["ingredients_to_exclude"]
-    recipes_number = query_dict["recipes_number"]
-    
-    #Retrieves recipes data matching user's query from Spoontacular API
-    recipes_data = get_spoontacular_data(ingredients_to_include, ingredients_to_exclude, recipes_number)
-    recipes_dict = recipes_data["results"]
-
-    #Builds a list of Recipe objects using the data obtained from Spoontacular API
-    recipes = [] 
-
-    for recipe in recipes_dict:
-        title = recipe["title"]
-        picture = recipe["image"]
-        ingredients = [ingredient["name"] for ingredient in recipe["nutrition"]["ingredients"]]
-        missing_ingredients_count = recipe["missedIngredientCount"]
-        carbs = recipe["nutrition"]["caloricBreakdown"]["percentCarbs"]
-        proteins = recipe["nutrition"]["caloricBreakdown"]["percentProtein"]
-        fats = recipe["nutrition"]["caloricBreakdown"]["percentFat"]
-        calories = recipe["nutrition"]["nutrients"][0]["amount"]
-        source_url = recipe["sourceUrl"]
-
-        recipe_obj = Recipe(title, picture, ingredients, missing_ingredients_count, carbs, proteins, fats, calories, source_url)
-        recipe_obj.get_translation()
-        recipes.append(recipe_obj)
-
-    #Renders page    
-    context = {
-        "recipes": recipes,
-    }
-    return render(request, "recipes/recipes_list.html", context=context)
+
+    def get_recipes(self, include, exclude, count, query):
```

Comment:
Naming conventions dictate that prepended `get` method should return something, however, this is returning None. `save` might be more fitting

Reply:
method refactured